app/src/pages/project/SessionsTable.tsx

```diff
@@ -0,0 +1,305 @@
+import { SpanFilterConditionField } from "./SpanFilterConditionField";
+import { spansTableCSS } from "./styles";
+
+type SessionsTableProps = {
+  project: SessionsTable_sessions$key;
+};
+
+const PAGE_SIZE = 50;
+
+export function SessionsTable(props: SessionsTableProps) {
+  //we need a reference to the scrolling element for logic down below
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const [sorting, setSorting] = useState<SortingState>([]);
+  const [filterCondition, setFilterCondition] = useState<string>("");
+  const navigate = useNavigate();
+  const { fetchKey } = useStreamState();
+  const { data, loadNext, hasNext, isLoadingNext, refetch } =
+    usePaginationFragment<SessionsTableQuery, SessionsTable_sessions$key>(
+      graphql`
+        fragment SessionsTable_sessions on Project
+        @refetchable(queryName: "SessionsTableQuery")
+        @argumentDefinitions(
+          after: { type: "String", defaultValue: null }
+          first: { type: "Int", defaultValue: 50 }
+        ) {
+          name
+          sessions(first: $first, after: $after, timeRange: $timeRange)
+            @connection(key: "SessionsTable_sessions") {
+            edges {
+              session: node {
+                id
+                sessionId
+                sessionUser
+                numTraces
+                startTime
+                endTime
+                firstInput {
+                  value
+                }
+                lastOutput {
+                  value
+                }
+                tokenUsage {
+                  prompt
+                  completion
+                  total
+                }
+              }
+            }
+          }
+        }
+      `,
+      props.project
+    );
+  const tableData = useMemo(() => {
+    return data.sessions.edges.map(({ session }) => session);
+  }, [data]);
+  type TableRow = (typeof tableData)[number];
+  const columns: ColumnDef<TableRow>[] = [
+    {
+      header: "session id",
+      accessorKey: "sessionId",
+      enableSorting: false,
+      cell: TextCell,
+    },
+    {
+      header: "session user",
+      accessorKey: "sessionUser",
+      enableSorting: false,
+      cell: TextCell,
+    },
+    {
+      header: "first input",
+      accessorKey: "firstInput.value",
+      enableSorting: false,
+      cell: TextCell,
+    },
+    {
+      header: "last output",
+      accessorKey: "lastOutput.value",
+      enableSorting: false,
+      cell: TextCell,
+    },
+    {
+      header: "start time",
+      accessorKey: "startTime",
+      enableSorting: false,
+      cell: TimestampCell,
+    },
+    {
+      header: "end time",
+      accessorKey: "endTime",
+      enableSorting: false,
+      cell: TimestampCell,
+    },
+    {
+      header: "total tokens",
+      accessorKey: "tokenUsage.total",
+      enableSorting: false,
+      minSize: 80,
+      cell: ({ row, getValue }) => {
+        const value = getValue();
+        if (value === null) {
```

Comment:
so you don't have to cast (`as number`) down below (should never happen, but safer) would suggest - switch null check to shallow equals since it will catch null or undefiend and check the type is a number

```suggestion
        if (value == null || typeof value !== "number") {
```